lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -799,6 +805,8 @@ private function doScheduleInsert($document, &$visited, $overrideIdGenerator = n
                 break;
             case self::STATE_DETACHED:
                 throw new InvalidArgumentException('Detached document or new document with already existing id passed to persist(): '.self::objToStr($document, $this->dm));
+            case self::STATE_FROZEN:
+            	throw new InvalidArgumentException('Versions of documents cannot be persisted: '.self::objToStr($document, $this->dm));
```

Comment:
Could refer to them as "Document versions" rather than "Versions of documents" ? Is that clearer?